src/renderers/common/ChainMap.js

```diff
@@ -30,9 +30,11 @@ export default class ChainMap {
 	 */
 	get( keys ) {
 
+		if ( keys.length === 0 ) return undefined;
```

Comment:
I don't think we need to cache the zero length case. This produce unnecessary overhead.

Reply:
Good, just realize that `weakmap.get( undefined )` returns undefined, instead of throwing error, so...

```js
map.get( keys[ keys.length - 1 ] ) // return undefined when keys.length === 0
```

👍